hs_communities/views/communities.py

```diff
@@ -0,0 +1,138 @@
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.utils.html import mark_safe, escapejs
+from django.views.generic import TemplateView
+
+from hs_access_control.management.utilities import community_from_name_or_id
+from hs_access_control.models.community import Community
+from hs_communities.models import Topic
+
+
+class CollaborateView(TemplateView):
+    template_name = 'pages/collaborate.html'
+
+
+class CommunityView(TemplateView):
+    template_name = 'hs_communities/community.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(CommunityView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        # user_id = User.objects.get(pk=self.request.user.id)
+        grpfilter = self.request.GET.get('grp')
+
+        community_resources = community_from_name_or_id("CZO National Community").public_resources
+        groups = []
+        for c in community_resources:
+            if not any(str(c.group_id) == g.get('id') for g in groups):  # if the group id is not already present in the list
+                if c.group_name != "CZO National":  # The National Group is used to establish the entire Community
+                    res_count = len([r for r in community_resources if r.group_name == c.group_name])
+                    groups.append({'id': str(c.group_id), 'name': str(c.group_name), 'res_count': str(res_count)})
+
+        groups = sorted(groups, key=lambda key: key['name'])
+        return {
+            'community_resources': community_resources,
+            'groups': groups,
+            'grpfilter': grpfilter,
+        }
+
+
+class FindCommunitiesView(TemplateView):
+    template_name = 'hs_communities/find-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(FindCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+
+        return {
+            'communities_list': Community.objects.all()
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class MyCommunitiesView(TemplateView):
+    template_name = 'hs_communities/my-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(MyCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def group_to_community(self, grp, communities):
+        """
+        return the community membership information of a group; group can belong to only one community
+        :param grp: group object
+        :param communities: communities
+        :return: tuple id, name of community
+        """
+        for community in communities:
+            if grp.id in [g.id for g in community.member_groups]:
+                return community
+
+    def get_context_data(self, **kwargs):
+        grps_member_of = []
+        u = User.objects.get(pk=self.request.user.id)
+        groups = Group.objects.filter(gaccess__active=True).exclude(name="Hydroshare Author")
```

Comment:
Hydroshare Author is hardcoded as an exclude? why?

Reply:
Shouldn't this query include user?
`Group.objects.filter(g2ugp__user=u, gaccess__active=True)`

Reply:
This is a pre-existing condition, wasn't looking to remove/break old functionality https://github.com/hydroshare/hydroshare/search?q=.exclude%28name%3D%22Hydroshare+Author%22%29&unscoped_q=.exclude%28name%3D%22Hydroshare+Author%22%29

Reply:
@sblack-usu @cuahsimarko I believe 'HydroShare Author" group is a pre-defined group from admin page to handle Mezzanine page permissions. Since it has nothing to do with created groups, it has been excluded from all group related queries from the beginning when Group functionality was implemented.